lib/server/services/lhcPeriod/LhcPeriodStatisticsService.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+const { repositories: { LhcPeriodStatisticsRepository } } = require('../../../database');
+const { dataSource } = require('../../../database/DataSource.js');
+const { ApiConfig } = require('../../../config');
+const { lhcPeriodStatisticsAdapter } = require('../../../database/adapters');
+
+/**
+ * @typedef LhcPeriodIdentifier object to uniquely identify a lhc period
+ * @property {string} [name] the lhc period name
+ * @property {number} [id] the id of lhc period
+ */
+
+/**
+ * LhcPeriodStatisticsService
+ */
+class LhcPeriodStatisticsService {
+    /**
+     * Find or create a lhc period model by its name or id
+     * @param {LhcPeriodIdentifier} identifier the criteria to find run type
+     * @return {Promise<SequelizeLhcPeriod[]>} the run type found or null
+     */
+    async getById({ id, name }) {
+        return lhcPeriodStatisticsAdapter.toEntity(await LhcPeriodStatisticsRepository.findOne({ id, name }));
+    }
+
+    /**
+     * Get all lhc periods with statistics
+     * @param {Object} dto definition of request
+     * @returns {CountedData} result
+     */
+    async getAllForPhysicsRuns(dto = {}) {
+        const { query = {} } = dto;
+        const { filter, page = {}, sort = { name: 'desc' } } = query;
+
+        const queryBuilder = dataSource.createQueryBuilder();
+
+        for (const property in sort) {
+            queryBuilder.orderBy(property, sort[property]);
+        }
+        const { limit = ApiConfig.pagination.limit, offset = 0 } = page;
+        queryBuilder.limit(limit);
+        queryBuilder.offset(offset);
+
+        if (filter) {
+            const { ids, names } = filter;
+            if (ids) {
+                queryBuilder.where('id').oneOf(...ids.split(ApiConfig.searchItemsSeparator));
```

Comment:
Use an array when you expect an array. Using string with specific separator is a very bad practice

Reply:
refactored